terraform/node_resource_apply.go

```diff
@@ -8,6 +8,45 @@ import (
 	"github.com/hashicorp/terraform/lang"
 )
 
+// nodeExpandApplyableResource handles the first layer of resource
+// expansion during apply. This is required because EvalTree does not have a
+// context with which to expand the resource into multiple instances.
+// This type should be a drop in replacement for NodeApplyableResource, and
```

Comment:
As a "drop-in replacement" is the implication that it is used instead of `NodeApplyableResource` in certain scenarios, but not all? And what are those scenarios? (something that might be helpful to clarify here)

Reply:
Ah good point, I do that too often in comments. I've already done the replacement, and the comment should only explain the "why"